ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -561,8 +572,11 @@ private void fetchRequests(RequestRegistrationCallback<GetOperation> requestRegi
     void checkAndMaybeComplete() {
       if (chunkOperationTracker.isDone()) {
         if (chunkOperationTracker.hasSucceeded()) {
-          // override any previously set exceptions.
-          chunkException = null;
+          // override any previously set exceptions as long as it's not a range out of bounds error,
+          // which is set after successfully reading a chunk.
+          if (chunkException != null && chunkException.getErrorCode() != RouterErrorCode.RangeNotSatisfiable) {
```

Comment:
would like this to be cleaned up a bit - if not we should add a comment about how as far as the operation tracker is concerned this is successful but the range satisfiability is outside of its scope. Would prefer if you can refactor the code to achieve that.